### PR TITLE
Feature/new remote control modules

### PR DESCRIPTION
**Pull request by BSolonenko**
Technical task : https://github.com/smartdevicelink/sdl_core/issues/1798

Implementation of proposals : 
[0099-new-remote-control-modules-and-parameters](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0099-new-remote-control-modules-and-parameters.md)
[0105-remote-control-seat](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0105-remote-control-seat.md)
[0106-remote-control-onRcStatus-notification](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0106-remote-control-onRcStatus-notification.md)

### CODE REVIEW

File: app/controller/MediaController.js [hunk 1]
```diff
@@ -245,36 +533,331 @@ SDL.MediaController = Em.Object.create(
...
+        result = SDL.deepCopy(resultNew);
+      }
+
+      return result;
+    },
+    getAudioControlCapabilities: function () {
+      var result = [];
+      var capabilities = {
+        moduleName: 'AudioControlCapabilities',
+        sourceAvailable: true,
```
Comment (yang1070) [inline app/controller/MediaController.js]: keepContextAvailable: true

File: app/controller/MediaController.js [hunk 2]
```diff
@@ -245,36 +533,331 @@ SDL.MediaController = Em.Object.create(
...
+        volumeAvailable: true,
+        equalizerAvailable: true,
+        equalizerMaxChannelId: 100
```
Comment (yang1070) [inline app/controller/MediaController.js]: 10

File: app/controller/MediaController.js [hunk 3]
```diff
@@ -35,6 +35,22 @@ SDL.MediaController = Em.Object.create(
     /**
      * Initial substate
      */
+    init: function () {
+      //this.tempEqualizerSettIndex=this.lastRadioControlStruct.equalizerSettings.length;
+      this.tempEqualizerSettIndex = 1;
+      this.tempEqualizerSettings.channelName = this.lastRadioControlStruct.equalizerSettings[this.tempEqualizerSettIndex - 1].channelName;
+      this.tempEqualizerSettings.channelId = parseInt(this.lastRadioControlStruct.equalizerSettings[this.tempEqualizerSettIndex - 1].channelId);
+      this.tempEqualizerSettings.channelSetting = parseInt(this.lastRadioControlStruct.equalizerSettings[this.tempEqualizerSettIndex - 1].channelSetting);
+      for (var i = 4; i <= 100; i++) {
```
Comment (yang1070) [inline app/controller/MediaController.js]: for (var i = 4; i <= 10; i++) {

File: app/controller/MediaController.js (hunk 1 above)
Comment (BSolonenko) [reply app/controller/MediaController.js]: @yang1070  Fixed in https://github.com/smartdevicelink/sdl_hmi/pull/52/commits/20972cea5d354553b9029450392e0493a26092d5

File: app/controller/MediaController.js (hunk 2 above)
Comment (BSolonenko) [reply app/controller/MediaController.js]: @yang1070  Fixed in https://github.com/smartdevicelink/sdl_hmi/pull/52/commits/20972cea5d354553b9029450392e0493a26092d5

File: app/controller/MediaController.js (hunk 3 above)
Comment (BSolonenko) [reply app/controller/MediaController.js]: @yang1070  Fixed in https://github.com/smartdevicelink/sdl_hmi/pull/52/commits/20972cea5d354553b9029450392e0493a26092d5

File: app/controller/MediaController.js [hunk 4]
```diff
@@ -43,41 +59,247 @@ SDL.MediaController = Em.Object.create(
...
-    turnOnRadio: function() {
+    turnOnRadio: function () {
+      this.deactivateAll();
       if (!SDL.States.media.player.radio.active) {
         SDL.States.goToStates('media.player.radio');
       }
       SDL.RadioModel.saveCurrentOptions();
       SDL.RadioModel.set('active', true);
+      this.returnParameters();
+      //SDL.SDLMediaController.deactivateActiveRcApp();
```
Comment (yang1070) [inline app/controller/MediaController.js]: why it is commented out?

File: app/controller/MediaController.js [hunk 5]
```diff
@@ -245,36 +533,333 @@ SDL.MediaController = Em.Object.create(
...
+              if (this.state) {
+                this.set('tempSource', 'MOBILE_APP');
+                this.set('state', false);
+                SDL.SDLController.onEventChanged('player', true);
+              }
+              switch (data.source) {
+                case 'RADIO_TUNER':
+                  this.setFalseStateModel();
+                  SDL.RadioModel.set('active', true);
+                  //SDL.States.set('media.player.radio.active',true);
```
Comment (yang1070) [inline app/controller/MediaController.js]: why keep the commented code?

File: app/controller/MediaController.js [hunk 6]
```diff
@@ -245,36 +533,333 @@ SDL.MediaController = Em.Object.create(
...
+                case 'MOBILE_APP':
+                  this.setFalseStateModel();
+                  SDL.SDLModel.set('data.limitedExist', true);
+                  //SDL.States.set('media.sdlmedia.active',true);
+                  this.set('activeState', 'media.sdlmedia'); this.set('state', true);
+                  SDL.SDLController.onEventChanged('player', false);
+                  this.set('tempSource', '');
+                  break;
+              }
+              //this.onPlayerEnter(SDL.BluetoothModel,'bluetooth');
```
Comment (yang1070) [inline app/controller/MediaController.js]: same question here?

File: app/model/media/BluetoothModel.js [hunk 7]
```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2013, Ford Motor Company All rights reserved.
```
Comment (yang1070) [inline app/model/media/BluetoothModel.js]: 2018

File: app/model/media/IPodModel.js [hunk 8]
```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2013, Ford Motor Company All rights reserved.
```
Comment (yang1070) [inline app/model/media/IPodModel.js]: 2018

File: app/model/media/LineInModel.js [hunk 9]
```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2013, Ford Motor Company All rights reserved.
```
Comment (yang1070) [inline app/model/media/LineInModel.js]: 2018

File: app/controller/MediaController.js [hunk 10]
```diff
@@ -245,36 +533,333 @@ SDL.MediaController = Em.Object.create(
...
+                case 'RADIO_TUNER':
+                  this.setFalseStateModel();
+                  SDL.RadioModel.set('active', true);
+                  //SDL.States.set('media.player.radio.active',true);
+                  this.set('activeState', 'media.player.radio');
+                  break;
+                case 'USB':
+                  this.setFalseStateModel();
+                  SDL.USBModel.set('active', true);
+                  //SDL.States.set('media.player.usb.active',true);
```
Comment (yang1070) [inline app/controller/MediaController.js]: same question here and in this file

File: app/view/media/audioView.js [hunk 11]
```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright (c) 2013, Ford Motor Company All rights reserved.
```
Comment (yang1070) [inline app/view/media/audioView.js]: 2018

File: app/view/settings/SeatView.js [hunk 12]
```diff
@@ -0,0 +1,1245 @@
...
+            contentBinding: 'SDL.SeatModel.enableStruct',
+            valueBinding: 'SDL.SeatModel.massageEnabledData',
+            change:function(){
+                SDL.SeatModel.set('tempSeatControlData.massageEnabled',
+            (SDL.SeatModel.massageEnabledData=='OFF')? true:false);
+            SDL.SeatModel.update();
+            }
+        })
+    }),
+    // massageEnable: Em.ContainerView.extend({
```
Comment (yang1070) [inline app/view/settings/SeatView.js]: do we still need  these?

File: app/controller/MediaController.js (hunk 4 above)
Comment (AKalinich-Luxoft) [reply app/controller/MediaController.js]: @yang1070 we have some commented code after some refactoring, now it is not needed
Removed in https://github.com/smartdevicelink/sdl_hmi/pull/52/commits/15df95ad8d27087a083f5af01f5dd66b32c12e2d

File: app/controller/MediaController.js (hunk 5 above)
Comment (AKalinich-Luxoft) [reply app/controller/MediaController.js]: @yang1070 we have some commented code after some refactoring, now it is not needed
Removed in 15df95a

File: app/controller/MediaController.js (hunk 6 above)
Comment (AKalinich-Luxoft) [reply app/controller/MediaController.js]: @yang1070 we have some commented code after some refactoring, now it is not needed
Removed in 15df95a

File: app/model/media/BluetoothModel.js (hunk 7 above)
Comment (AKalinich-Luxoft) [reply app/model/media/BluetoothModel.js]: @yang1070 fixed in 15df95a

File: app/model/media/IPodModel.js (hunk 8 above)
Comment (AKalinich-Luxoft) [reply app/model/media/IPodModel.js]: @yang1070 fixed in 15df95a

File: app/model/media/LineInModel.js (hunk 9 above)
Comment (AKalinich-Luxoft) [reply app/model/media/LineInModel.js]: @yang1070 fixed in 15df95a

File: app/controller/MediaController.js (hunk 10 above)
Comment (AKalinich-Luxoft) [reply app/controller/MediaController.js]: @yang1070 we have some commented code after some refactoring, now it is not needed
Removed in 15df95a

File: app/view/media/audioView.js (hunk 11 above)
Comment (AKalinich-Luxoft) [reply app/view/media/audioView.js]: @yang1070 fixed in 15df95a

File: app/view/settings/SeatView.js (hunk 12 above)
Comment (AKalinich-Luxoft) [reply app/view/settings/SeatView.js]: @yang1070 don't need anymore
Removed in 15df95a

**Comment by yang1070** [comment]
@theresalech  Ford has reviewed, tested and approved this PR, It is ready for Livio to review.

**Comment by JackLivio** [comment]
@BSolonenko Hi Would it be possible to split this PR into three separate PR's based on their corresponding evolution proposals?

**Comment by AStasiuk** [comment]
Ad-hoc testing of feature related manual tests is  successful.
@yang1070 could you please review provided changes?

**Comment by yang1070** [comment]
@AStasiuk When I send RPC to change the audio source to `RAIDO_TUNNER` while HMI was in FM radio, it didn't show mobile app's template
screenshot_2018-08-16_15-24-43

**Comment by yang1070** [comment]
@AStasiuk  When the RC access mode is set to `ASK_DRIVER`, and I send a SetInteriorVehicleData request to turn on a light with app1, and then anther request with same content with app2, it show the pop up. The info in the pop up is not correct. Module shall be `LIGHT` instead of unknown.
screenshot_2018-08-16_15-47-45

**Comment by AStasiuk** [comment]
@yang1070 reported issues are under fixing now, I would like to stress, that are only HMI issues, which are not affected the SDL Core code.

**Comment by yang1070** [comment]
@AStasiuk I see the issues are fixed. I'm glad it does not affect SDL core code.

**Pull request merged**